### ISSUE TITLE
Unwanted completion when hitting backspace

### ISSUE THREAD

**Issue by danielRi** [nushell/nushell#5497]
### Discussed in https://github.com/nushell/nushell/discussions/5495

<div type='discussions-op-text'>

<sup>Originally posted by **danielRi** May 10, 2022</sup>
I noticed that sometimes I get code completions where I´m not expecting them. I try to describe my problem here. Im not sure, maybe some people will say "this is expected behavior, and its good", but for me it seems it shouldnt be like this.

You can see an occasion of the unwanted completion of a folder name in the video. Here is what I´m doing in the video:

1. typing `cd aa`
2. hitting `tab`
3. there are no folders/files that start with `aa`, so the prompt correctly shows `NO RECORDS FOUND` - everything good so far
4. Now Im hitting `backspace`, deleting one a:

**Observed, unwanted behavior:**
The auto completion triggers, and I now have the whole folder name in my prompt input. I need to delete it completely, even though i just typed `aa`, which irritates me at this point (especially if I dont expect it and was hacking commands very quickly). I know I could delete the word quickly with hotkeys, but it´s still not preferred behavior.

**Expected:**

I would expected that I just have `a` in my input prompt, I can just delete it and then type the name from start again

https://user-images.githubusercontent.com/3048681/167540617-ea28013d-15f7-4674-85ed-98397b1af53b.mov

</div>

Environment

| key                | value                               |
| ------------------ | ----------------------------------- |
| version            | 0.62.0                              |
| branch             |                                     |
| tag                |                                     |
| build_os           | macos-aarch64                       |
| rust_version       | rustc 1.60.0 (7737e0b5c 2022-04-04) |
| rust_channel       | stable-aarch64-apple-darwin         |
| cargo_version      | cargo 1.60.0 (d1fd9fe2c 2022-03-01) |
| pkg_version        | 0.62.0                              |
| build_time         | 2022-05-10 02:54:33 +02:00          |
| build_rust_channel | release                             |
| features           | default, trash, which, zip          |
| installed_plugins  |                                     |

**Comment by fdncred** [comment]
That is using `quick_completions`. If you set that to false, it wouldn't complete when you start backspacing. 

I was thinking this was about how it lists every file on your system sometimes when you get down to one letter. That's similar and annoying too.

**Comment by danielRi** [comment]
@fdncred I agree. I also noticed the listing of all files sometimes, havent completely identified where that came from yet. Thanks for pointing it out. `quick_completions false` also disables this?

**Comment by fdncred** [comment]
This is why i have this keybinding.
```
     {
       name: clear
       modifier: none
       keycode: esc
       mode: emacs
       event: { edit: clear }
    }
```
so i can hit `esc` and it clears the command line

**Comment by fdncred** [comment]
> @fdncred I agree. I also noticed the listing of all files sometimes, havent completely identified where that came from yet. Thanks for pointing it out. `quick_completions false` also disables this?

Yes, adding `quick_completions: false` disables it.

as documented in the default_config.nu
```
    quick_completions: true  # set this to false to prevent auto-selecting completions when only one remains
```
